src/mrpro/operators/_SliceProjectionOp.py

```diff
@@ -0,0 +1,449 @@
+        self.fwhm = torch.as_tensor(fwhm)
+
+    def __call__(self, x: torch.Tensor) -> torch.Tensor:
+        return torch.exp(-(x**2) / (0.36 * self.fwhm**2))
+
+
+class SliceSmoothedRect:
+    """Rectangular Slice Profile with smooth flanks.
+
+    The smaller n, the smoother it is. For n<1, the FWHM might be wrong
+
+    """
+
+    def __init__(self, fwhm: float | Tensor, n: float | Tensor):
+        self.n = n
+        self.fwhm = fwhm
+
+    def __call__(self, x: torch.Tensor) -> torch.Tensor:
+        y = x * 2 / self.fwhm
+        return torch.erf(self.n * (1 - y)) + torch.erf(self.n * (1 + y))
+
+
+class SliceInterpolate:
+    """Slice Profile based on Interpolation of Measured Profile."""
+
+    def __init__(self, xs: Tensor, weights: Tensor):
+        self._xs = xs
+        self._weights = weights
+
+    def __call__(self, x: torch.Tensor) -> torch.Tensor:
+        return torch.as_tensor(np.interp(x, self._xs.numpy(), self._weights.numpy(), 0, 0))
+
+
+TensorFunction: TypeAlias = Callable[[Tensor], Tensor]
+DefaultGaussianSliceProfile = SliceGaussian(2.0)
+
+
+class SliceProjectionOp(LinearOperator):
+    matrix: torch.Tensor | None
+    matrix_adjoint: torch.Tensor | None
+
+    def __init__(
+        self,
+        input_shape: SpatialDimension[int],
+        slice_rotation: Rotation | None = None,
+        slice_shift: float | Tensor = 0.0,
+        slice_profile: TensorFunction | np.ndarray | NestedSequence[TensorFunction] | float = 1.0,
+        optimize_for: Literal['forward', 'adjoint', 'both'] = 'both',
+    ):
+        """Create a module that represents the 'projection' of a volume onto a plane.
+
+        This operation samples from a 3D Volume a slice with a given rotation and shift,
+        according to the slice_profile. It can, for example, be used to describe the slice
+        selection of a 2D MRI sequence from the 3D Volume.-
+
+        The projection will be done by sparse matrix multiplication.
+
+        Rotation, shift, and profile can have (multiple) batch dimensions. These dimensions will
+        be broadcasted to a common shape and added to the front of the volume.
+        Different settings for different volume batches are NOT supported, consider creating multiple
+        operators if required,
+
+
+        Parameters
+        ----------
+        input_shape:
+            Shape of the 3D volume to sample from
+        slice_rotation
+            Rotation that describes the orientation of the plane. If None,
+            an identity rotation is used.
+        slice_shift
+            Offset of the plane in the volume perpendicular plane from the center of the volume.
+        slice_profile:
+            A function that called with a distance x from the slice center should return the
+            intensity along the slice thickness at x. This can also be a nested Sequence or an
+            numpy array of functions.
+            If it is a single float, it will be interpreted as a Gaussian FWHM.
```

Comment:
This seems unintuative to me. If a function is set to a constant value, I would epect it to return a constant value rather than a Gaussian profile.

Reply:
This is way it is explicitly describe  in the docstring.
Or rename the parameter? 

We could also remove the float option  here, but I found it  useful to be able to only specify a width and have a default...

Reply:
Are do you mean a rectangular slice profile with this width?

Reply:
Describing something in a docstring is a good option but IMO it is better to provide someting more intuative such that the user doesn't even  have to read the docstring

Reply:
so rectangluar profile. I have no opionion on the default profile shape, so a hard rect is as good as a  gaussian for me.
> This seems unintuative to me. If a function is set to a constant value, I would epect it to return a constant value rather than a Gaussian profile.

--> this is still a issue with the rect profile.
the slice_profilce function for a rectangle would be.. a rectangle. not a constant value. so the issue that it is either 'profile' function, or it is a width float still exists.  a constant function would be an infinite width, uniform profile.
Not sure if there is a better way to do it though if we want to have the option of only setting a width (which I prefer)

Reply:
> Not sure if there is a better way to do it though if we want to have the option of only setting a width (which I prefer)

I can't think of a better way. As far as I understand you are normalising your slice profile anyway so a "a constant function would be an infinite width" would always be the same independet of setting this value to 1 or 100. A rectangular profile seems to be more useful then.